sched/definitions.go

```diff
@@ -24,3 +24,26 @@ type Command struct {
 	// We may extend this with more configuration,
 	// E.g. environment variables
 }
+
+// Status for Job & Tasks
+type Status int
+
+const (
+	// NotRunning, waiting to be scheduled
+	NotStarted Status = iota
+
+	// Currently Scheduled and In Progress Job/Task
+	InProgress
+
+	// Successfully Completed Job/Task
+	Completed
+
+	// Job was Aborted, Compensating Tasks are being Applied.
+	// A RollingBack task has not finished its compensating
+	// tasks yet.
+	RollingBack
```

Comment:
It's asymmetric wrt the forward-case: there we have ready, running, done. But here we seem to have only one state for ready/running.

Not a huge deal, especially since rollback isn't our roadmap (but then I wonder why we want to handle it now, instead of just panicking because ahhh we don't put sagas into this state at all).